android/beagle/src/main/java/br/com/zup/beagle/android/components/TabBar.kt

```diff
@@ -0,0 +1,144 @@
+import br.com.zup.beagle.android.utils.observeBindChanges
+import br.com.zup.beagle.android.view.ViewFactory
+import br.com.zup.beagle.android.widget.RootView
+import br.com.zup.beagle.android.widget.WidgetView
+import br.com.zup.beagle.annotation.RegisterWidget
+import br.com.zup.beagle.core.Style
+import br.com.zup.beagle.widget.core.Flex
+import com.google.android.material.tabs.TabLayout
+
+private val TAB_BAR_HEIGHT = 48.dp()
+
+@RegisterWidget
+data class TabBar(
+    val items: List<TabBarItem>,
+    val styleId: String? = null,
+    val currentTab: Bind<Int>? = null,
+    val onTabSelection: List<Action>? = null
+) : WidgetView() {
+
+    @Transient
+    private val viewFactory: ViewFactory = ViewFactory()
+
+    override fun buildView(rootView: RootView): View {
+        val containerFlex = Style(flex = Flex(grow = 1.0))
+        val tabBar = makeTabLayout(rootView.getContext())
+        val container = viewFactory.makeBeagleFlexView(rootView.getContext(), containerFlex)
+        configTabSelectedListener(tabBar, rootView)
+        configCurrentTabObserver(tabBar, rootView)
+        container.addView(tabBar)
+        return container
+    }
+
+    private fun makeTabLayout(context: Context) : TabLayout = viewFactory.makeTabLayout(
+        context,
+        styleManagerFactory.getTabViewStyle(styleId)
+    ).apply {
+        layoutParams =
+            viewFactory.makeFrameLayoutParams(
+                FrameLayout.LayoutParams.MATCH_PARENT,
+                TAB_BAR_HEIGHT
+            )
+        tabMode = TabLayout.MODE_SCROLLABLE
+        tabGravity = TabLayout.GRAVITY_FILL
+        configTabBarStyle()
+        addTabs(context)
+    }
+
+    private fun TabLayout.configTabBarStyle() {
+        styleManagerFactory.getTabBarTypedArray(context, styleId).apply {
+            setSelectedTabIndicatorColor(
+                getColor(
+                    R.styleable.BeagleTabBarStyle_tabIndicatorColor,
+                    styleManagerFactory.getTypedValueByResId(R.attr.colorAccent, context).data
+                )
+            )
+            tabIconTint = getColorStateList(R.styleable.BeagleTabBarStyle_tabIconTint)
+            recycle()
+        }
+    }
+
+    private fun TabLayout.addTabs(context: Context) {
+        for (i in items.indices) {
+            addTab(newTab().apply {
+                text = items[i].title
+                items[i].icon?.let {
+                    icon = getIconFromResources(context, it.mobileId)
+                }
+            })
+        }
+    }
+
+    private fun getIconFromResources(context: Context, icon: String): Drawable? {
+        return BeagleEnvironment.beagleSdk.designSystem?.image(icon)?.let {
+            ContextCompat.getDrawable(context, it)
+        }
+    }
+
+    private fun configTabSelectedListener(tabBar: TabLayout, rootView: RootView) {
+        tabBar.addOnTabSelectedListener(object : TabLayout.OnTabSelectedListener {
+            override fun onTabSelected(p0: TabLayout.Tab?) {
```

Comment:
What do you think about changing the parameter name?
```suggestion
            override fun onTabSelected(tab: TabLayout.Tab?) {
```